go/cmd/betabot/main.go

```diff
@@ -103,15 +101,71 @@ func handleEvent(ctx context.Context, messengerClient bertymessenger.MessengerSe
 			if err != nil {
 				return err
 			}
+		} else if contact.State == bertymessenger.Contact_Established {
+			// When contact was established, send message and a group invitation
+			<-time.After(5 * time.Second)
+			userMessage, err := proto.Marshal(&bertymessenger.AppMessage_UserMessage{
+				Body: "Hey! Welcome to the beta!",
+			})
+			if err != nil {
+				return err
+			}
+			_, err = messengerClient.Interact(ctx, &bertymessenger.Interact_Request{
+				Type:                  bertymessenger.AppMessage_TypeUserMessage,
+				Payload:               userMessage,
+				ConversationPublicKey: contact.ConversationPublicKey,
+			})
+			if err != nil {
+				return err
+			}
+			<-time.After(5 * time.Second)
```

Comment:
That still sync (a chan is async but interacting with it (`<-`) is sync), what is different with chan is you can use them in selects:
```go
select {
case value:=<-channel:
  // code executed when a value was red from the channel
case channel<-valueToWrite:
  // code executed when a value was written from the channel
case <-time.After(time.Second):
  // code executed if nothing was done after 1s
}
```
When a select is done it will first try to to do a non blocking interact with the channels one by one starting from the top (non blocking means if can't interact (due to channel blocked or buffer less) skip the case), if a channel interaction was sucessfull it will execute the case and exit the select with the implied break statement at the end of select and switch case in go.
Then if all non blocking interaction failed the runtime takes the lead, basicaly your goroutine will likely context switch and will be eventualy be wakeup later on a sucessfull case but you can see this as quite random (if your first case would be executable first and then a second one your goroutine might be resumed later with the second or first case).

Note that if you only want the first part of select (checking all cases one by one) but not the runtime sleepy part (waiting until a case resolve) you can add a default case in your select :
```go
select {
case value:=<-channel:
  // code executed when a value was red from the channel
case channel<-valueToWrite:
  // code executed when a value was written from the channel
default:
  // code executed if all other interaction were blocking (we rarely need to put code there but it might be usefull)
}
// Next code
```